iwyu_output.cc

```diff
@@ -1844,6 +1897,27 @@ size_t IwyuFileInfo::CalculateAndReportIwyuViolations() {
   // we *do* need to add it.
   set<string> associated_desired_includes = AssociatedDesiredIncludes();
 
+  // Resolve using declarations before continuing.  This handles the case
+  // where there's a using dclaration in the file but no code is actually
+  // using it. If that happens, we might try to remove all of the headers with
+  // the decls that the using decl is referencing, which would result in a
+  // compilation error at best. A possible solution is to remove the using
+  // decl if it's not used, but that doesn't work for header files because a
+  // using decl in a header is an exported symbol, so we don't want to do that
+  // either. As a compromise, we arbitrarily add the first shadow decl to make
+  // sure everything still compiles instead of removing the using decel. A
+  // more thorough approach would be to scan the current list of includes that
+  // alredy name this decl (like in the overloaded function case) and include
+  // one of those so we don't include a file we dont actually need.
+  for(auto&& using_line : using_decl_lines_) {
+    if(!using_line.is_referenced()) {
+      const UsingDecl* using_decl = using_line.using_decl();
+      ReportForwardDeclareUse(using_decl->getUsingLoc(),
+                              using_decl->shadow_begin()->getTargetDecl(), 
+                              false, nullptr);
```

Comment:
I'll put it on my todo list to find these and clean them up.

The more "solutions" I see to the lack of named arguments in C++, the more I like the lo-fi comment employed by LLVM & Clang. It's cheap and gets the job done (until the argument is renamed, of course :-))